File: engine/model/src/main/scala/tools/packet/statusinfo/ShowMesoGain.scala
package tools.packet.statusinfo

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class ShowMesoGain(private var _gain: Int, private var _inChat: Boolean) extends PacketInput {
  def gain: Int = _gain

  def inChat: Boolean = _inChat

  override def opcode(): SendOpcode = SendOpcode.SHOW_STATUS_INFO
}

File: engine/engine-model/src/main/java/tools/packet/statusinfo/ShowMesoGain.java
package tools.packet.statusinfo;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record ShowMesoGain(Integer gain, Boolean inChat) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.SHOW_STATUS_INFO;
   }
}